components/script/dom/htmlimageelement.rs

```diff
@@ -871,6 +871,166 @@ impl HTMLImageElement {
+                match response {
+                    // TODO Handle multipart/x-mixtape
+                    Ok(ImageOrMetadataAvailable::ImageAvailable(_image, _url)) => {}
+                    Ok(ImageOrMetadataAvailable::MetadataAvailable(_m)) => { }
+                    Err(ImageState::Pending(_id)) => { }
+                    Err(ImageState::NotRequested(_id)) => { }
+                    Err(ImageState::LoadError) => {
+                        *self.pending_request.borrow_mut() = ImageRequest {
+                            state: State::Unavailable,
+                            parsed_url: None,
+                            source_url: None,
+                            image: None,
+                            metadata: None,
+                            blocker: None,
+                            final_url: None,
+                            current_pixel_density: None,
+                        };
+                    }
+
+                }
+            }
+
+            let this = Trusted::new(self);
+            let new_selected_source = src.to_string();
+            let relevant_mutation = self.generation.get() != *generation;
+            // Step 15
+            let _ = window.dom_manipulation_task_source().queue(
+                task!(image_load_event: move || {
+                            let this = this.root();
+                            // Step 15.1
+                            if relevant_mutation {
+                                let mut _pending_request =
+                                    &*this.pending_request.borrow_mut();
+                                    _pending_request = &ImageRequest {
+                                    state: State::Unavailable,
+                                    parsed_url: None,
+                                    source_url: None,
+                                    image: None,
+                                    metadata: None,
+                                    blocker: None,
+                                    final_url: None,
+                                    current_pixel_density: None,
+                                };
+                                return;
+                            }
+
+                            // Step 15.2
+                            *this.last_selected_source.borrow_mut() = Some(DOMString::from_string(new_selected_source));
+                            let mut current_request = this.current_request.borrow_mut();
+                            current_request.current_pixel_density = selected_pixel_density;
+
+                            // Step 15.3
+                            this.current_request.borrow_mut().state = State::CompletelyAvailable;
+
+                            // Step 15.4
+                            // Already a part of the list of available images due to Step 14
+
+                            // Step 15.5
+                            current_request = this.pending_request.borrow_mut();
```

Comment:
This doesn't actually overwrite the contents of `current_request`. We need `*current_request` for that.